dask/utils.py

```diff
@@ -1111,6 +1111,17 @@ def partial_by_order(*args, **kwargs):
 def is_arraylike(x):
     """Is this object a numpy array or something similar?
 
+    This function tests specifically for an object that already has
+    array attributes (e.g. np.ndarray, dask.array.Array), **NOT** for
+    something that can be coerced into an array object (e.g. Python
+    lists and tuples). It is meant for dask developers and developers
+    of downstream libraries.
+
+    Note that this function is inconsistent with NumPy's definition of
```

Comment:
This has a slightly more positive connotation

```suggestion
    Note that this function does not correspond with NumPy's definition of
```

Reply:
Agreed :+1: